utils/notification.ts

```diff
@@ -142,6 +145,53 @@ const successNH = (): NHResultSuccess =>
     kind: "SUCCESS"
   });
 
+/** Check if an Azure ServiceBus Dictionary is empty */
+const dictionaryIsEmpty = (
+  obj: Azure.ServiceBus.Dictionary<string | unknown>
+): boolean => Object.keys(obj).length === 0;
+
+/** Compose the error message coming from NH */
+const composeNHErrorMessage = (
+  err: Error,
+  response: Azure.ServiceBus.Response
+): string =>
+  fromNullable(response)
+    .chain(res =>
+      some(
+        !res.body || dictionaryIsEmpty(res.body) ? "No response body" : res.body
+      )
+    )
+    .map(innerMessage =>
+      innerMessage
+        ? `[error message: ${err.message}] [response: ${response.statusCode} - ${innerMessage}]`
+        : `[error message: ${err.message}] [response: ${response.statusCode}]`
+    )
+    .getOrElse(`[error message: ${err.message}]`);
+
+/**
+ * Handle the Azure Notification Hub callback and
+ * return either an Error or a `NHResultSuccess`
+ *
+ * @param err The error message from Notification Hub, if any
+ * @param response The response from Notification Hub
+ * @returns either an Error or a success result
+ */
+const handleResponseOrError = (
+  err: Error | null,
+  response: Azure.ServiceBus.Response
+): Either<Error, NHResultSuccess> =>
+  err == null
+    ? either.of(successNH())
+    : left(new Error(composeNHErrorMessage(err, response)));
```

Comment:
you can use `fromNullable` from `Either`

```ts
import { fromNullable as fromNullableE } from "fp-ts/lib/Either"

fromNullableE(err)
  .bimap(
    _ => new Error(composeNHErrorMessage(err, response)),
    _ => successNH()
  )
```

so you don't have to ask yourself "what if it's just _undefined_?"

Reply:
But in this case, if err is not null, it will be treated as `right`? so should we invert the logic?